cfd_protocol/src/lib.rs

```diff
@@ -209,12 +208,157 @@ pub fn spending_tx_sighash(
 }
 
 pub fn finalize_spend_transaction(
-    _tx: Transaction,
-    _commit_descriptor: Descriptor<PublicKey>,
-    (_maker_pk, _maker_sig): (PublicKey, Signature),
-    (_taker_pk, _taker_sig): (PublicKey, Signature),
-) -> Transaction {
-    todo!()
+    tx: Transaction,
+    spent_descriptor: &Descriptor<PublicKey>,
+    (maker_pk, maker_sig): (PublicKey, Signature),
+    (taker_pk, taker_sig): (PublicKey, Signature),
+) -> Result<Transaction> {
+    let satisfier = {
+        let mut satisfier = HashMap::with_capacity(2);
+
+        satisfier.insert(maker_pk, (maker_sig, SigHashType::All));
+        satisfier.insert(taker_pk, (taker_sig, SigHashType::All));
+
+        satisfier
+    };
```

Comment:
Might be able to condense this using the `from_iter` function on `HashMap`: https://doc.rust-lang.org/std/collections/struct.HashMap.html#impl-FromIterator%3C(K%2C%20V)%3E

```rust
HashMap::from_iter(vec![
	(maker_pk, (maker_sig, SigHashType::All)),
	(taker_pk, (taker_sig, SigHashType::All))
]);
```